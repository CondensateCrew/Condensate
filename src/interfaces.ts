// DATA INTERFACES
export interface UserLoginPosting {
  email: string,
  password: string
}

export interface UserLoginReceived {
  readonly id: number,
  readonly firstName: string,
  readonly lastName: string
}

export interface UserSignupPosting {
  email: string,
  password: string,
  firstName: string,
  lastName: string,
  id?: number
}

export interface Action {
  id: number,
  name: string
}

export interface Brainstorm {
  id: number,
  question: string,
  idea: string,
  action: string,
  isGenius: boolean,
  categories: Category[]
}

export interface Category {
  id: number,
  name: string
}

export interface Insight {
  id: number,
  question: string,
  answers: string[]
}

export interface WordContext {
  type: string,
  connectedWord: string[]
}

export interface RandomWordCollection {
  [key: string]: WordContext
}

export interface AppStore {
  user: UserLoginReceived,
  allBrainstorms: Brainstorm[],
  actions: Action[],
  categories: Category[],
  readonly randomWordCollections: RandomWordCollection[],
  readonly questionTemplates: string[],
  readonly secretSauce: string[],
  currentBrainstorm: Brainstorm,
  chosenWords: string[],
  insights: Insight[],
  query: string,
  filter: string,
  timeEnded: boolean,
  instructionsEnded: boolean
}

// ACTIONS OBJECTS

export interface ActionObject {
  type: string,
  [key: string]: any
}

export interface IAddUserAction {
  type: 'ADD_USER',
  user: UserLoginReceived
}

export interface ILogoutUserAction {
  type: 'LOG_OUT_USER'
}

export interface IAddAllActionsAction {
  type: 'ADD_ALL_ACTIONS',
  actions: Action[]
}

export interface IAddNewActionAction {
  type: 'ADD_NEW_ACTION',
  action: Action
}

export interface IAddAllBrainstormsAction {
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: Brainstorm[]
}

export interface IChangeBrainstormAction {
  type: 'TOGGLE_BRAINSTORMS',
  id: number
}

export interface IAddNewBrainstormAction {
  type: 'ADD_BRAINSTORM',
  brainstorm: Brainstorm
}

export interface IDeleteBrainstormAction {
  type: 'DELETE_BRAINSTORM',
  id: number
}

export interface IAddAllCategoriesAction {
  type: 'ADD_ALL_CATEGORIES',
  categories: Category[]
}

export interface IAddNewCategoryAction {
  type: 'ADD_CATEGORY',
  category: Category
}

export interface IAddRandomWordCollectionsAction {
  type: 'ADD_ALL_COLLECTIONS',
  collections: RandomWordCollection[]
}

export interface IAddQuestionTemplatesAction {
  type: 'ADD_ALL_TEMPLATES',
  templates: string[]
}

export interface IAddSecretSauceAction {
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: string[]
}

export interface IAddCurrentBrainstormAction {
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: Brainstorm
}

export interface IRemoveCurrentBrainstormAction {
  type: 'REMOVE_CURRENT_BRAINSTORM'
}

export interface IAddChosenWordAction {
  type: 'ADD_WORD',
  chosenWord: string
}

export interface IRemoveChosenWordAction {
  type: 'REMOVE_WORD',
  chosenWord: string
}

export interface IAddInsightAction {
  type: 'ADD_INSIGHT',
  insight: Insight
}

export interface IAddQueryAction {
  type: 'ADD_QUERY',
  query: string
}

export interface IRemoveQueryAction {
  type: 'REMOVE_QUERY'
}

export interface IAddFilterAction {
  type: 'ADD_FILTER',
  filter: string
}

export interface IRemoveFilterAction {
  type: 'REMOVE_FILTER'
}

export interface IEndTimeAction {
  type: 'END_TIME'
}

export interface IReverseTimeAction {
  type: 'REVERSE_TIME'
}

export interface IEndInstructionsAction {
  type: 'END_INSTRUCTIONS'
}

export interface IReverseInstructionsAction {
  type: 'REVERSE_INSTRUCTIONS'
}
