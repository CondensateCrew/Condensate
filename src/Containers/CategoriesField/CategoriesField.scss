@import '../../css/index.scss';
@import '../../css/variables.scss';

.category-field-section {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  padding: $extra-small-space;
}

.add-category-btn {
  width: 1.75rem;
  height: 1.75rem;
  padding: .2rem;
  background: $primary-gradient;
  border: $light-border;
  border-radius: $round-border-radius;
  box-shadow: $external-shadow;
  &:hover {
    background: $hover-btn-gradient;
  }
}
