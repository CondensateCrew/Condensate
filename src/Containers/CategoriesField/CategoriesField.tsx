import React from 'react';
import './CategoriesField.scss';
import BrainstormCategory from '../../Components/BrainstormCategory/BrainstormCategory';
import mockCategories from '../../data/mockCategories';
// import addIcon from '../../assets/add.svg';
import { IBrainstormForm, Category } from '../../interfaces';

interface Props {
  formState: IBrainstormForm,
  setCategory: (formState: IBrainstormForm) => void
};

const CategoryField:React.FC<Props>= ({ formState, setCategory }) => {
  const categories: JSX.Element[] = mockCategories.map((category: Category, index:number) => (
    <BrainstormCategory
      key={index}
      category={category}
      formState={formState}
      setCategory={setCategory}/>
  ));

  return (
    <section className='category-field-section'>
      {
        // <img className='add-category-btn' alt='add-icon' src={addIcon}/>
      }
      {categories}
    </section>
  );
}

export default CategoryField;
