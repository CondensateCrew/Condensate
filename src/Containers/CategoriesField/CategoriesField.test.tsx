import React from 'react';
import { shallow } from 'enzyme';
import { IBrainstormForm, Category } from '../../interfaces';
import CategoriesField from './CategoriesField';

const mockCategory: Category[] = [
  {id: 1, name: 'Education'},
  {id: 2, name: 'Technology'},
  {id: 3, name: 'Environment'},
  {id: 4, name: 'Food'},
  {id: 5, name: 'Music'}
];

jest.mock('react-redux', () => ({
  useSelector: () => mockCategory
}));

describe('CategoriesField', () => {
  let wrapper: any;
  const mockFormState: IBrainstormForm = {
    question: '',
    categories: [],
    action: {
      id: 1,
      action: 'create an app'
    },
    reset: true
  };

  const mockSetCategory = jest.fn();
  const mockProps = {
    formState: mockFormState,
    setCategory: mockSetCategory
  };

  beforeEach(() => {
    wrapper = shallow(<CategoriesField {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
