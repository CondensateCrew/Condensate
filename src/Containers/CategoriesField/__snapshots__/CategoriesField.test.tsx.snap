// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesField should match the snapshot 1`] = `
<section
  className="category-field-section"
>
  <BrainstormCategory
    category={
      Object {
        "id": 1,
        "name": "Education",
      }
    }
    formState={
      Object {
        "action": "create an app",
        "categories": Array [],
        "question": "",
        "reset": true,
      }
    }
    key="0"
    setCategory={[MockFunction]}
  />
  <BrainstormCategory
    category={
      Object {
        "id": 2,
        "name": "Technology",
      }
    }
    formState={
      Object {
        "action": "create an app",
        "categories": Array [],
        "question": "",
        "reset": true,
      }
    }
    key="1"
    setCategory={[MockFunction]}
  />
  <BrainstormCategory
    category={
      Object {
        "id": 3,
        "name": "Environment",
      }
    }
    formState={
      Object {
        "action": "create an app",
        "categories": Array [],
        "question": "",
        "reset": true,
      }
    }
    key="2"
    setCategory={[MockFunction]}
  />
  <BrainstormCategory
    category={
      Object {
        "id": 4,
        "name": "Food",
      }
    }
    formState={
      Object {
        "action": "create an app",
        "categories": Array [],
        "question": "",
        "reset": true,
      }
    }
    key="3"
    setCategory={[MockFunction]}
  />
  <BrainstormCategory
    category={
      Object {
        "id": 5,
        "name": "Music",
      }
    }
    formState={
      Object {
        "action": "create an app",
        "categories": Array [],
        "question": "",
        "reset": true,
      }
    }
    key="4"
    setCategory={[MockFunction]}
  />
</section>
`;
